src/pods/configure-verbs/configure-verbs.component.tsx

```diff
@@ -15,30 +15,68 @@ interface Props {
   onCancel: () => void;
 }
 
-export const ConfigureVerbsComponent: React.FC<Props> = props => {
-  const { verbCollection, onSave, onCancel } = props;
-  const [temporalSelection, setTemporalSelection] = React.useState(
-    verbCollection
-  );
+const useTemporalSelectionAndSelected = (verbCollection: VerbEntity[]) => {
+  const [temporalSelection, setTemporalSelection] = React.useState<VerbEntity[]>([]);
   const [temporalSelected, setTemporalSelected] = React.useState(0);
```

Comment:
By temporalSelected, you mean how many items are selected? Is that equales to temporalSelection.length?

Could you please ellaborate a bit more? Thanks

Reply:
I see, the list can contain a different number of items selected, I need to understand a bit deeper how this logic works, could you please ellaborate?

Reply:
Ok, I explain it.
_**temporalSelection**_ has always the same quantity of elements, only the field _**selected**_ on the elements changes.
So I needed another state to store how many verbs are selected at the moment. In a first approach, I tried to calculate this number using function **getSelectedNumber** (_configure-verbs.business.ts_) without having the state. But this is inefficient, it is better to call **getSelectedNumber**  once, when the component is mounted.
Why do I need the currect number of select verbs? For the master checkbox (when _**temporalSelection**_ = 0, is unchecked; when >= 0 & <= total, is indeterminate; when = total, is checked)

Reply:
Mmmm... we could take advantage of the hooks and the ability to execute some code just whenever a variable changes, something like

```tsx
// calculateRootCheckBoxState
React.useEffect(() => {
  // Do your calculation thing, is zero? is equals to length? ...
  // set the checkbox status
}, [temporalSelection,]);
```
By doing this you only execute this code whenever temporalSelection gets updated and you save having temporary variables.

What do you think?